### DISCUSSION TITLE
Access eleventy configuration in shortcodes

### DISCUSSION THREAD

**Discussion by JbIPS** [11ty/eleventy#4113] [Enhancement Queue]
Some shortcodes would greatly benefits from accessing eleventy configuration, like [`eleventy-img` with `pathprefix`](https://github.com/11ty/eleventy-img/issues/44).

In the same way we could [access `page.data`](https://www.11ty.dev/docs/languages/nunjucks/#access-to-page-data-values) in shortcodes, exposing a `this.config` could help tweak shortcode depending on the global configuration of Eleventy.

At the moment, this can be achieved by bypassing CLI configuration with `env` variables and settings everything in the JS. Providing a clean way to access config would be more straightforward.

<details>
<summary>Workaround example:</summary>

```js   
const Path = require('path');
const fs = require("fs");
const Image = require('@11ty/eleventy-img');

module.exports = function(eleventyConfig) {
  const pathPrefix = process.env.PATH_PREFIX || '/';

  eleventyConfig.addNunjucksShortcode("image", (src, alt, sizes, classes, widths) => {
    const options = {
      widths: widths ? widths.split(' ').map(s => parseInt(s, 10)) : [300],
      formats: ["webp"],
      outputDir: '_site/img',
      urlPath: pathPrefix + 'img/'
    };

    Image(src, options);

    const metadata = Image.statsSync(src, options);

    let imageAttributes = {
      alt,
      sizes,
      loading: "lazy",
      decoding: "async",
      class: `img-fluid ${classes || ''}`
    };

    return Image.generateHTML(metadata, imageAttributes);
  })

  return {
    pathPrefix
  };
};
```

</details>

**Comment by zachleat** [comment]
I think there is a lot of pieces to this request but just to loop the pathprefix point together:

The new HTML `<base>` plugin solves the path prefixing feature in a much cleaner way: https://www.11ty.dev/docs/plugins/html-base/ and won’t require any plugin changes to apply path prefix throughout your project
